src/reducers/profile-view.js

```diff
@@ -665,6 +661,10 @@ export const selectorsForThread = (
       getRangeSelectionFilteredTracingMarkers,
       (markers): TracingMarker[] => markers.filter(tm => tm.name !== 'GCMajor')
     );
+    const getMarkerTiming = createSelector(
+      getRangeSelectionFilteredTracingMarkers,
+      MarkerTiming.getMarkerTiming
+    );
```

Comment:
I believe this is the important change: we loop over marker timings in the chart. Even if we don't draw all of them, if we have a lot of markers merely looping through them can be costly.

The obvious drawback is that we'll have to regenerate the marker timings at each range change. I'd like that you measure the effect of changing the range on this. Maybe it's less costly to filter the marker timings using the range, rather than rebuild the marker timings from the markers filtered with the range (if you follow me). But I really don't know so we need to measure.

Reply:
But for sure this is better to do this filtering here, in memoized selectors, rather than do it again and again in the react component.